src/state.rs

```diff
@@ -25,11 +25,12 @@ impl AppState {
 #[cfg(test)]
 impl AppState {
     pub fn for_testing(pool: atmosphere::Pool) -> Self {
-        let config_path_str = format!("/tmp/gitlab-runner-config-{}.toml", uuid::Uuid::new_v4());
-        Self {
-            pool,
-            config_path: PathBuf::from(config_path_str),
-        }
+        let config_path = PathBuf::from(format!(
```

Comment:
This makes one config file per code execution, right? Is this for testing purposes?

I imagine that this uuid will later be matchable to the issuer?

Reply:
yes, this is for testing purposes (impl block is marked `#[cfg(test)]` so will only be compiled for tests.

the reason for this is that Rust tests execute in arbitrary order, and they execute in parallel. UUID might be overkill here but it does make it very improbable that we get collisions in tests because the same file name was used repeatedly.